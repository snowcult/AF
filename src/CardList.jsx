import Card from "./Card.jsx";

const CardList = ({ robots }) => {
  const cardComponent = robots.map((user, i) => {
    return <Card key={i} id={user.id} name={user.name} email={user.email} />;
  });
  return <div>{cardComponent}</div>;
};

export default CardList;
