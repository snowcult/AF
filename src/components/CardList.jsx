import Card from './Card.jsx';
import PropTypes from 'prop-types';
const CardList = ({ robots }) => {
  return (
    <div className="tc">
      {robots.map((item, i) => (
        <span key={i}>
          <Card id={item.id} name={item.name} email={item.email} />;
        </span>
      ))}
    </div>
  );
};
CardList.propTypes = {
  robots: PropTypes.array.isRequired,
};
export default CardList;
