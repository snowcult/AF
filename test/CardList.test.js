import { shallow } from 'enzyme';
import React from 'react';
import CardList from '../src/components/CardList.jsx';

it('expect to render CardList comp', () => {
  const mockRobots = [{ id: 1, name: 'jon', username: 'ss', email: 'san@si' }];
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
