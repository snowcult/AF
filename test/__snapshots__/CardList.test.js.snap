// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList comp 1`] = `
<div
  className="tc"
>
  <span
    key="0"
  >
    <Card
      email="san@si"
      id={1}
      name="jon"
    />
    ;
  </span>
</div>
`;
